config/azure.go

```diff
@@ -2,13 +2,109 @@ package config
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"net/url"
+	"strings"
 
+	"github.com/databricks/databricks-sdk-go/apierr"
 	"github.com/databricks/databricks-sdk-go/httpclient"
 	"github.com/databricks/databricks-sdk-go/logger"
 	"golang.org/x/oauth2"
 )
 
+type tokenError struct {
+	message string
+	err     *httpclient.HttpError
+}
+
+func (e *tokenError) Error() string {
+	return e.message
+}
+
+func (e *tokenError) Unwrap() error {
+	sdkErr, ok := apierr.ByStatusCode(e.err.StatusCode)
+	if ok {
+		// this is how we distinguish between bad requests and permission denies
+		return sdkErr
+	}
```

Comment:
Am I understanding correctly that this maps responses from AAD/ARM onto our error codes? Do we depend on this somewhere?

Reply:
that's correct - if we set a wrong AAD tenant ID, client code would expect to handle `apierr.ErrBadRequest` - no matter if it's our API or AAD's API, same for `apierr.ErrUnauthorized`.